commons/utils.js

```diff
@@ -66,6 +67,115 @@ const getUnit = bridgeMode => {
   return { unitHome, unitForeign }
 }
 
+const parseValidatorEvent = event => {
+  if (
+    event.event === undefined &&
+    event.raw &&
+    event.raw.topics &&
+    (event.raw.topics[0] === '0xe366c1c0452ed8eec96861e9e54141ebff23c9ec89fe27b996b45f5ec3884987' ||
+      event.raw.topics[0] === '0x8064a302796c89446a96d63470b5b036212da26bd2debe5bec73e0170a9a5e83')
+  ) {
+    const rawAddress = event.raw.topics.length > 1 ? event.raw.topics[1] : event.raw.data
+    const address = '0x' + rawAddress.slice(26)
+    event.event = 'ValidatorAdded'
+    event.returnValues.validator = address
+  } else if (
+    event.event === undefined &&
+    event.raw &&
+    event.raw.topics &&
+    event.raw.topics[0] === '0xe1434e25d6611e0db941968fdc97811c982ac1602e951637d206f5fdda9dd8f1'
+  ) {
+    const rawAddress = event.raw.data === '0x' ? event.raw.topics[1] : event.raw.data
+    const address = '0x' + rawAddress.slice(26)
+    event.event = 'ValidatorRemoved'
+    event.returnValues.validator = address
+  }
+}
+
+const processValidatorsEvents = events => {
+  const validatorList = new Set()
+  events.forEach(event => {
+    parseValidatorEvent(event)
+
+    if (event.event === 'ValidatorAdded') {
+      validatorList.add(event.returnValues.validator)
+    } else if (event.event === 'ValidatorRemoved') {
+      validatorList.delete(event.returnValues.validator)
+    }
+  })
+
+  return Array.from(validatorList)
+}
+
+const tryCall = async (method, fallbackValue) => {
+  try {
+    return await method.call()
+  } catch (e) {
+    return fallbackValue
+  }
+}
+
+const getDeployedAtBlock = async contract => tryCall(contract.methods.deployedAtBlock(), 0)
+
+const getPastEvents = async (contract, { event, fromBlock, toBlock, options }) => {
+  let events
+  try {
+    events = await contract.getPastEvents(event || 'allEvents', {
+      options: options || {},
```

Comment:
`options` is not a valid parameter for `getPastEvents`. The correct way will be to add content of `options` object directly. See example [here](https://web3js.readthedocs.io/en/v1.2.0/web3-eth-contract.html#id41). One posible fix could be:
```javascript
const getPastEvents = async (contract, { event, fromBlock, toBlock, options = {} }) => {
  let events
  try {
    events = await contract.getPastEvents(event || 'allEvents', {
      ...options,
      fromBlock,
      toBlock: toBlock || 'latest'
    })
```

Reply:
Right, my mistake. Updated.